src/Giveaway.js

```diff
@@ -534,30 +555,41 @@ class Giveaway extends EventEmitter {
                     .replace('{winners}', formattedWinners)
                     .replace('{prize}', this.prize)
                     .replace('{messageURL}', this.messageURL);
-                const channel =
-                    (this.message.channel.isThread() && !this.message.channel.sendable &&
-                    !this.message.channel.permissionsFor(this.client.user)?.has([
-                        this.message.channel.locked ? 'MANAGE_THREADS' : 'SEND_MESSAGES',
-                        this.message.channel.type === 'GUILD_PRIVATE_THREAD'
-                            ? 'USE_PRIVATE_THREADS'
-                            : 'USE_PUBLIC_THREADS',
-                    ]))
-                        ? this.message.channel.parent
+                    
+                const channel = 
+                    (
+                        (this.libraryIsEris ? [10, 11, 12].includes(this.message.channel.type) : this.message.isThread()) &&
+                        (this.libraryIsEris ? 
+                            !(!this.message.channel.threadMetadata.archived && 
+                                (this.message.channel.type !== 12 || this.message.channel.member || this.message.channel.permissionsOf(this.client.user.id).has('MANAGE_THREADS')) &&
+                                this.message.channel.permissionsOf(this.client.user.id).has(this.message.channel.threadMetadata.locked ? 'MANAGE_THREADS' : 'SEND_MESSAGES') &&
+                                this.message.channel.permissionsOf(this.client.user.id).has(this.message.channel.type === 12 ? 'USE_PRIVATE_THREADS' : 'USE_PUBLIC_THREADS')
+                            )
+                            : !this.message.channel.sendable && !this.message.channel.permissionsFor(this.client.user)?.has([
+                                this.message.channel.locked ? 'MANAGE_THREADS' : 'SEND_MESSAGES',
+                                this.message.channel.type === 'GUILD_PRIVATE_THREAD' ? 'USE_PRIVATE_THREADS' : 'USE_PUBLIC_THREADS',
+                            ]))
+                    ) ? this.libraryIsEris
+                            ? await this.message.channel.guild.channels.find(async (c) => [0, 5].includes(c.type) && (await c.getMessage(this.message.channel.parentID)))
```

Comment:
Is there no other way to get a message without providing the channel?

Reply:
I don't think so.